### PR TITLE
backpacks now catch fire

### PR DESCRIPTION
**Pull request by toomykins**
[tweak]
because muh immulshions.
I may come to regret this one.
image

Here is a video to demonstrate how it works at this stage since nobody seems to be able to figure it out
https://www.youtube.com/watch?v=I7PLrtgfZyM

:cl:
 * rscadd: Putting hot things into your backpack now makes it and you catch fire. Just stop drop and roll.

### CODE REVIEW

**Comment by dunsel-hordon** [comment]
but wait does this affect BoH? Cause if so i think a BoH burning up should have some pretty bad consequences.

**Comment by toomykins** [comment]
> but wait does this affect BoH? Cause if so i think a BoH burning up should have some pretty bad consequences.

look at second commit, it's the exception

**Comment by BomberBro** [comment]
so if the backpack burns up all the items inside burn too? does this ash items inside (like an o2 tank) or just set flammable ones on fire?

**Comment by Chucklemonger** [comment]
Neat!
Does this occur to toolbelts too? Muh welders and all

File: code/ZAS/Fire.dm [hunk 1]
```diff
@@ -58,13 +58,21 @@ Attach to transfer valve and open. BOOM.
...
+		var/obj/O = src
+		O.source_temperature = initial(O.source_temperature)
 
 /atom/proc/ignite(var/temperature)
 	on_fire=1
 	//visible_message("\The [src] bursts into flame!")
 	if(fire_dmi && fire_sprite)
 		fire_overlay = image(fire_dmi,fire_sprite)
 		overlays += fire_overlay
+	if(istype(src,/obj))
```
Comment (Exxion) [inline code/ZAS/Fire.dm]: >typechecking `src`

Don't do this. Put it in `/obj/ignite()`

File: code/ZAS/Fire.dm [hunk 2]
```diff
@@ -58,13 +58,21 @@ Attach to transfer valve and open. BOOM.
...
+		var/obj/O = src
+		O.source_temperature = temperature
+		var/datum/gas_mixture/A = O.loc.return_air()
+		O.loc.fire_act(A,temperature,A.return_volume())
```
Comment (Exxion) [inline code/ZAS/Fire.dm]: It's just `A.volume`. `return_volume()` is the old way to do it.

File: code/game/objects/items/weapons/storage/backpack.dm [hunk 3]
```diff
@@ -14,12 +14,22 @@
...
 /obj/item/weapon/storage/backpack/attackby(obj/item/weapon/W as obj, mob/user as mob)
 	if(!stealthy(user))
 		playsound(src, "rustle", 50, 1, -5)
 	. = ..()
 
+
+/obj/item/weapon/storage/backpack/ashify()
+	var/datum/gas_mixture/A = loc.return_air()
+	for(var/obj/item/O in contents)
+		O.fire_act(A,src.is_hot(),A.return_volume())
```
Comment (Exxion) [inline code/game/objects/items/weapons/storage/backpack.dm]: Again, just `A.volume`

File: code/game/objects/items/weapons/storage/storage.dm [hunk 4]
```diff
@@ -341,6 +341,8 @@
 		usr.update_icons()
 	W.forceMove(src)
 	W.on_enter_storage(src)
+	var/datum/gas_mixture/A = loc.return_air()
+	src.fire_act(A,W.is_hot(),A.return_volume())
```
Comment (Exxion) [inline code/game/objects/items/weapons/storage/storage.dm]: Ditto

File: code/ZAS/Fire.dm (hunk 1 above)
Comment (toomykins) [reply code/ZAS/Fire.dm]: thanks, do i need a ..() in there?

Comment (toomykins) [reply code/ZAS/Fire.dm]: that's a yes

**Comment by toomykins** [comment]
Thanks for the feedback exxion, much appreciated.

>Neat!
>Does this occur to toolbelts too? Muh welders and all

Not at this moment, toolbelts are stronk leather after all.

>so if the backpack burns up all the items inside burn too? does this ash items inside (like an o2 tank) or just set flammable ones on fire?

Most items melt. I'll code boxes to eject their items before they're turned into ash.

**Comment by BomberBro** [comment]
bad idea then, you should be able to at least stop drop and roll to get the fire out on the bag since sinks and showers aren't exactly common in many places. it doesn't make sense that only backpacks burn either when everything else, including my paper hat, are unaffected by fire, especially when EVERYTHING in the bag burns into ash including metal, welding tools, and other flame resistant material. I could see this if like, fire melting stuff was more common but as it stands it just negatively affects honest, regular joes who don't rush bags of holding

**Comment by toomykins** [comment]
you're probably right
maybe i should put this on hold til i've figured out how to get everything just right.

**Comment by SonixApache** [comment]
is this finally it? the return of the slag?

**Comment by toomykins** [comment]
you know i only make bad PR's

**Comment by gurfan** [comment]
I like the idea but it should only destroy flammable things in your bag, or maybe even just the bag itself

**Comment by toomykins** [comment]
i couldn't get it to work as to where it burns your backpack by plasmafire if you have it on.
it only burns if you take it off and drop it in a very hot environment (same temperature at which wood burns)
or if you put something in it.
the thing you put in it has a temperature, which is the deciding factor for what items melt. 
If you keep your bag on, and don't put a flaming lighter or a red hot welding tool in it, you can safely walk through plasmafire.

**Comment by githubuser4141** [comment]
It would make sense for BoHs to be fire resistant considering they must be extremely tough to not blow apart since they're miniature black holes

**Comment by SageOfLight** [comment]
So wait this would destroy items in the bag regardless of material or destructibility in other situations involving fire?

If it can't burn outside of the bag in a plasma fire why would it work inside one. People would literally just use this as a disposable incinerator.

**Comment by toomykins** [comment]
> So wait this would destroy items in the bag regardless of material or destructibility in other situations involving fire?
> 
> If it can't burn outside of the bag in a plasma fire why would it work inside one.

i already answered this.
it depends on the heat of the thing you put in the bag to make it catch fire.
if it's hot enough to burn wood, it will burn the bag.
if it's in the bag and it's burnable and the temperature is hot enough for it to burn, it will burn.
if it's in the bag and itt's meltable and the temperature is hot enough for it to melt, it will melt.

**Comment by SonixApache** [comment]
I would support this but only if you have plans to expand this to other systems.
Punishing people for putting a lit welder in their backpack (yeah, I saw that thread post) is very very gay.

also another yell about SLAG ADD SLAG RETURN OF THE SLAG and so on

**Comment by toomykins** [comment]
>would support this but only if you have plans to expand this to other systems.

such as?

>Punishing people for putting a lit welder in their backpack (yeah, I saw that thread post) is very very gay.

why?

>also another yell about SLAG ADD SLAG RETURN OF THE SLAG and so on

I don't even know what slag is.

**Comment by SonixApache** [comment]
>such as?

such as making other things melt, making the autoignition temperature for objects actually work, fires happening more often and being more meaningfull, etcetera, etcetera

>why

because it's a very specific thing to target, and something that generally only happens by accident. I don't see the benefit from this, I don't see why this in particular should be punished

A lot of it would make more sense if this was the first step towards making fires more common.

>slag

#1832 and related

**Comment by bathosbathtime** [comment]
This is an oversight patch to a theoretical overhaul of our fire system, not a step forward onto itself. I don't know why would anyone want this with our current system but it is at least impressive for what it is.

**Comment by toomykins** [comment]
you're both practically asking for an overhaul to the fire system. I don't see much wrong with the current fire system, the best we could do is add more cases where things can catch fire. I'd like it if wooden/carpet floortiles could catch and spread fire, but simply dropping a lighter onto a carpet seems too easy. Truth is most areas don't have a lot of flammable material.
if i were to add a lot more to the fire system then it should probably be done in a different PR, since this is a whole subject onto itself.

**Comment by bathosbathtime** [comment]
I also don't see much wrong with the current fire system but I can also be within proximity of hot objects, even picking them up, with little consequence. Yet when I set them in my backpack, they suddenly have consequences for handling. I find that idea to be strange, though I also don't handle hot objects in game too often so I may be mistaken on how they work currently.

**Comment by toomykins** [comment]
seeing as it is rather unpopular and unrefined in other areas, i'll close this.

**Pull request closed without merging**